nucypher/blockchain/eth/interfaces.py

```diff
@@ -262,12 +264,7 @@ def get_blocktime(self):
 
     @property
     def is_connected(self) -> bool:
-        """
-        https://web3py.readthedocs.io/en/stable/__provider.html#examples-using-automated-detection
-        """
-        if self.client is NO_BLOCKCHAIN_CONNECTION:
-            return False
-        return self.client.is_connected
+        return self.__is_connected
```

Comment:
For the same reason as my previous comment, I wouldn't call this variable `__is_connected`, but maybe something like `__is_initialized`. In fact, I think this is the crux of the original issue. We're overloading the semantics of the term "is_connected", since we're currently assigning two meanings to it: one to check if an interface is initialized or set up, and another to check if a connection is alive. I would separate both concepts in two different methods; in this case, one to check if the interface was initialized (via the private initialization variable), and the other to probe the blockchain connectivity (via the `w3.is_connected` method).

Reply:
✔️